src/macros.rs

```diff
@@ -215,20 +235,18 @@ fn rewrite_macro_inner(
     if ts.is_empty() && !has_comment {
         return match style {
             Delimiter::Parenthesis if position == MacroPosition::Item => {
-                Some(format!("{macro_name}();"))
-            }
-            Delimiter::Bracket if position == MacroPosition::Item => {
-                Some(format!("{macro_name}[];"))
+                Ok(format!("{macro_name}();"))
             }
-            Delimiter::Parenthesis => Some(format!("{macro_name}()")),
-            Delimiter::Bracket => Some(format!("{macro_name}[]")),
-            Delimiter::Brace => Some(format!("{macro_name} {{}}")),
+            Delimiter::Bracket if position == MacroPosition::Item => Ok(format!("{macro_name}[];")),
+            Delimiter::Parenthesis => Ok(format!("{macro_name}()")),
+            Delimiter::Bracket => Ok(format!("{macro_name}[]")),
+            Delimiter::Brace => Ok(format!("{macro_name} {{}}")),
             _ => unreachable!(),
         };
     }
     // Format well-known macros which cannot be parsed as a valid AST.
     if macro_name == "lazy_static!" && !has_comment {
-        if let success @ Some(..) = format_lazy_static(context, shape, ts.clone()) {
+        if let success @ Ok(..) = format_lazy_static(context, shape, ts.clone(), mac.span()) {
             return success;
         }
     }
```

Comment:
```rust
        match format_lazy_static(context, shape, ts.clone(), mac.span()) {
            Ok(rw) => return Ok(rw),
            Err(err) => match err {
                RewriteError::MacroFailure { macro_error_kind, span: _ } 
                if macro_error_kind== MacroErrorKind::ParseFailure => {},
                _ => return Err(err),
            },
        }
```
I think we can improve error reporting on calling `format_lazy_static` this way. 

If we fail to parse macro with known (expected) syntax, we should still need to try formatting like following example. 
```tests/source/lazy_static.rs
lazy_static!(xxx, yyyy, zzzzz)
```

However, if we already succeeds to parse `lazy_static` with known syntax but fails to format due to other reason, it's better to tell users about it. 

I want to hear your opinion about this possible improvement.

Reply:
Interesting idea. I think we can give this a try!